cmd/chronograf-migrator/dashboard.go

```diff
@@ -0,0 +1,381 @@
+		}
+	case "line-stepplot":
+		v.Properties = influxdb.XYViewProperties{
+			Queries:    convertQueries(cell.Queries),
+			Axes:       convertAxes(cell.Axes),
+			Type:       "xy",
+			Legend:     convertLegend(cell.Legend),
+			Geom:       "step",
+			ViewColors: convertColors(cell.CellColors),
+			Note:       cell.Note,
+			Position:   "overlaid",
+		}
+	case "bar":
+		v.Properties = influxdb.XYViewProperties{
+			Queries:    convertQueries(cell.Queries),
+			Axes:       convertAxes(cell.Axes),
+			Type:       "xy",
+			Legend:     convertLegend(cell.Legend),
+			Geom:       "bar",
+			ViewColors: convertColors(cell.CellColors),
+			Note:       cell.Note,
+			Position:   "overlaid",
+		}
+	case "line-plus-single-stat":
+		v.Properties = influxdb.LinePlusSingleStatProperties{
+			Queries:    convertQueries(cell.Queries),
+			Axes:       convertAxes(cell.Axes),
+			Legend:     convertLegend(cell.Legend),
+			ViewColors: convertColors(cell.CellColors),
+			Note:       cell.Note,
+			Position:   "overlaid",
+		}
+	case "single-stat":
+		v.Properties = influxdb.SingleStatViewProperties{
+			Queries:    convertQueries(cell.Queries),
+			ViewColors: convertColors(cell.CellColors),
+			Note:       cell.Note,
+			// TODO(desa): what to do about ShowNoteWhenEmpty?
+		}
+	case "gauge":
+		v.Properties = influxdb.GaugeViewProperties{
+			Queries:    convertQueries(cell.Queries),
+			ViewColors: convertColors(cell.CellColors),
+			Note:       cell.Note,
+			// TODO(desa): what to do about ShowNoteWhenEmpty?
+		}
+	case "table":
+		v.Properties = influxdb.TableViewProperties{
+			Queries:    convertQueries(cell.Queries),
+			ViewColors: convertColors(cell.CellColors),
+			//TableOptions
+			//FieldOptions
+			Note: cell.Note,
+			// TODO(desa): what to do about ShowNoteWhenEmpty?
+		}
+	case "note":
+		v.Properties = influxdb.MarkdownViewProperties{
+			Note: cell.Note,
+		}
+	case "alerts", "news", "guide":
+		// TODO(desa): these do not have 2.x equivalents
+		v.Properties = influxdb.EmptyViewProperties{}
+	default:
+		v.Properties = influxdb.EmptyViewProperties{}
+	}
+
+	c.View = &v
+
+	return c
+}
+
+func convert1To2Variable(t chronograf.Template) (influxdb.Variable, error) {
+	v := influxdb.Variable{
+		Description: t.Label,
+		Name:        t.Var[1 : len(t.Var)-1], // trims `:` from variables prefix and suffix
+	}
+
+	switch t.Type {
+	case "influxql", "databases", "fieldKeys", "tagKeys", "tagValues", "measurements":
+		if t.Query == nil {
+			return v, fmt.Errorf("expected template variable to have non-nil query")
+		}
+	}
+
+	switch t.Type {
+	case "influxql":
+		v.Arguments = &influxdb.VariableArguments{
+			Type: "query",
+			Values: influxdb.VariableQueryValues{
+				Query:    fmt.Sprintf("// %s", t.Query.Command),
+				Language: "flux",
+			},
+		}
+	case "databases":
+		v.Arguments = &influxdb.VariableArguments{
+			Type: "query",
+			Values: influxdb.VariableQueryValues{
+				Query:    fmt.Sprintf("// SHOW DATABASES %s", t.Query.DB),
```

Comment:
question out of my own ignorance, what are the `//` for?

Reply:
Just to make the query a comment